provider/internal/builtin/service_client.go

```diff
@@ -623,21 +651,50 @@ func runOSSpecificGrepCommand(pattern string, location string, providerContext p
 		// escape other chars used in perl pattern
 		escapedPattern = strings.ReplaceAll(escapedPattern, "'", "'\\''")
 		escapedPattern = strings.ReplaceAll(escapedPattern, "$", "\\$")
-		cmd := fmt.Sprintf(
-			`find %v -type f -exec perl -ne 'print "$ARGV:$.:$1\n" if /%v/; close ARGV if eof;' {} +`,
-			location, escapedPattern,
-		)
+		cmd := ""
+		if ok, paths := providerContext.GetScopedFilepaths(); ok {
+			cmd = fmt.Sprintf(
+				`echo '%s' | \
+			xargs perl -ne '/%v/ && print "$ARGV:$.:$1\n";'`,
```

Comment:
Just curious if `xargs perl` or `find` on following lines behave as we expect on other plaforms like Mac (or also Windows?).

Reply:
perl is only for Mac, notice the if condition. We now have tests that run on all three platforms expecting identical output for same tcs